projects/bp-strapi/src/extensions/users-permissions/passwordRequirements.ts

```diff
@@ -0,0 +1,51 @@
+import {
+  validateChangePasswordBody,
+  validateRegisterBody,
+  validateResetPasswordBody,
+} from '@strapi/plugin-users-permissions/server/controllers/validation/auth';
+import { StrapiContext } from '../../types';
+import { getAuthController, wrap } from './helper';
+
+const minimumPasswordLength = 12;
+const minimumCharacterClasses = [/[a-z]/, /[A-Z]/, /[0-9]/, /[^a-zA-Z0-9]/];
+
+const fulfillsPasswordRequirements = (password: string) => {
+  if (password.length < minimumPasswordLength) {
+    return false;
+  }
+  if (!minimumCharacterClasses.every(characterClass => password.match(characterClass))) {
+    return false;
+  }
+  return true;
+};
+
+export const enforcePasswordRequirements = (password: string) => {
+  if (!fulfillsPasswordRequirements(password)) {
+    throw new Error(
+      `password must be at least ${minimumPasswordLength} characters long ` +
+        'and must contain at least a small letter, a capital letter, a number and a special characeter'
```

Comment:
character please

Reply:
also, should we consider translating this error message?

Reply:
> also, should we consider translating this error message?

They already are:
https://github.com/hpi-swa-lab/BP2021RH1/blob/911f66375518763dc9f3aa2c98ab74f08667577f/projects/bp-gallery/src/shared/locales/de.json#L98-L117
https://github.com/hpi-swa-lab/BP2021RH1/blob/911f66375518763dc9f3aa2c98ab74f08667577f/projects/bp-gallery/src/i18n.ts#L23-L27